portfolio/src/main/java/com/google/sps/servlets/SubmitEmailServlet.java

```diff
@@ -19,12 +26,26 @@
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
     // Get the value entered in the form.
-    String email = request.getParameter("email-address");
+    // String email = request.getParameter("email-address");
+
+    String email = Jsoup.clean(request.getParameter("email-address"), Whitelist.none());
+    long timestamp = System.currentTimeMillis();
+
+    Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+    KeyFactory keyFactory = datastore.newKeyFactory().setKind("Email");
+    FullEntity taskEntity =
+        Entity.newBuilder(keyFactory.newKey())
```

Comment:
Is this key complete? Do we need a key based on the user details to retrieve data ? Or else data will be overwritten with multiple calls right? 


https://googleapis.dev/java/google-cloud-datastore/latest/com/google/cloud/datastore/KeyFactory.html#KeyFactory-java.lang.String-java.lang.String-

Reply:
The tutorial used newKey( ) and getKey( ) with no parameters, but do you suggest that I change newKey to have a parameter of the user's email?

Reply:
I added a parameter to newKey() function call

Reply:
I see. Sorry, you can follow the tutorial.

Wonder how you would retrieve all emails if you submit multiple emails since the key is not unique.

Reply:
After testing my code, I found that by using newKey() with no parameters, a unique key is automatically generated for each new request. So for the first email submitted, it was assigned a key of 1, and the next email submitted was assigned a key of 2, and so on.